amqp/src/main/java/io/github/tcdl/msb/adapters/amqp/AmqpMessageProcessingTask.java

```diff
@@ -34,10 +33,13 @@ public AmqpMessageProcessingTask(String consumerTag, String body, Channel channe
     public void run() {
         try {
             LOG.debug(String.format("[consumer tag: %s] Starting message processing: %s", consumerTag, body));
-            msgHandler.onMessage(body);
+            msgHandler.onMessage(body, ackHandler);
             LOG.debug(String.format("[consumer tag: %s] Message has been processed: %s", consumerTag, body));
+            ackHandler.autoConfirm();
```

Comment:
Good point. Solution is to use setAutoAcknowledgement(false) if message processing need to be continued out of current thread.